testsuite/MDAnalysisTests/coordinates/test_gro.py

```diff
@@ -187,107 +188,136 @@ def __init__(self):
 
 
 class TestGROReader(BaseReaderTest):
-    def __init__(self, reference=None):
-        if reference is None:
-            reference = GROReference()
-        super(TestGROReader, self).__init__(reference)
+    @staticmethod
+    @pytest.fixture()
+    def ref():
+        return GROReference()
+
+    @staticmethod
+    @pytest.fixture()
+    def reader(ref):
+        reader = ref.reader(ref.trajectory)
+        reader.add_auxiliary('lowf', ref.aux_lowf, dt=ref.aux_lowf_dt, initial_time=0, time_selector=None)
+        reader.add_auxiliary('highf', ref.aux_highf, dt=ref.aux_highf_dt, initial_time=0, time_selector=None)
+        return reader
 
     def test_flag_convert_lengths(self):
         assert_equal(mda.core.flags['convert_lengths'], True,
                      "MDAnalysis.core.flags['convert_lengths'] should be True "
                      "by default")
 
-    def test_time(self):
-        u = mda.Universe(self.ref.topology, self.ref.trajectory)
+    def test_time(self, ref, reader):
+        u = mda.Universe(ref.topology, ref.trajectory)
         assert_equal(u.trajectory.time, 0.0,
                      "wrong time of the frame")
 
-    def test_full_slice(self):
-        u = mda.Universe(self.ref.topology, self.ref.trajectory)
+    def test_full_slice(self, ref, reader):
+        u = mda.Universe(ref.topology, ref.trajectory)
         trj_iter = u.trajectory[:]
         frames = [ts.frame for ts in trj_iter]
         assert_equal(frames, np.arange(u.trajectory.n_frames))
 
 
 class TestGROWriter(BaseWriterTest):
-    def __init__(self, reference=None):
-        if reference is None:
-            reference = GROReference()
-        super(TestGROWriter, self).__init__(reference)
-        self.u_no_resnames = make_Universe(['names', 'resids'],
-                                           trajectory=True)
-        self.u_no_resids = make_Universe(['names', 'resnames'],
-                                         trajectory=True)
-        self.u_no_names = make_Universe(['resids', 'resnames'],
+
+    @staticmethod
+    @pytest.fixture()
+    def ref():
+        return GROReference()
+
+    @staticmethod
+    @pytest.fixture()
+    def reader(ref):
+        return ref.reader(ref.trajectory)
+
+    @staticmethod
+    @pytest.fixture()
+    def u_no_resnames():
+        return make_Universe(['names', 'resids'], trajectory=True)
+
+    @staticmethod
+    @pytest.fixture()
+    def u_no_resids():
+        return make_Universe(['names', 'resnames'], trajectory=True)
+
+    @staticmethod
+    @pytest.fixture()
+    def u_no_names():
+        return make_Universe(['resids', 'resnames'],
                                         trajectory=True)
 
-    def test_write_velocities(self):
-        u = mda.Universe(self.ref.topology, self.ref.trajectory)
-        outfile = self.tmp_file('write-velocities-test')
+    @staticmethod
+    @pytest.fixture()
+    def tmpdir():
+        return tempdir.TempDir()
```

Comment:
Oh man overwriting the pytest tmpdir fixture isn't a good idea. rather call it `tempdir` to make it distinctively different. Also this fixture shouldn't be repeated for every class that inherits BaseTestWriter. Can't it be part of the base class?